torchvision/csrc/cpu/ROIAlign_cpu.cpp

```diff
@@ -309,9 +313,13 @@ void ROIAlignBackward(
     T roi_end_w = offset_rois[3] * spatial_scale - offset;
     T roi_end_h = offset_rois[4] * spatial_scale - offset;
 
-    // Force malformed ROIs to be 1x1
-    T roi_width = std::max(roi_end_w - roi_start_w, (T)1.);
-    T roi_height = std::max(roi_end_h - roi_start_h, (T)1.);
+    T roi_width = roi_end_w - roi_start_w;
+    T roi_height = roi_end_h - roi_start_h;
+    if (aligned) {
```

Comment:
!aligned

Reply:
Thanks for spotting this, just sent a commit fixing this.

This originally happened because I started with a different implementation, but then changed my mind in the middle and forgot to re-update the condition...

Reply:
This makes me think that I should add a test with degenerate boxes for RoIAlign with `aligned=False`.